File: src/test/java/com/dotnomi/fabricdependencyinjection/java/testclasses/identifier/DataProcessor.java
package com.dotnomi.fabricdependencyinjection.java.testclasses.identifier;

import com.dotnomi.fabricdependencyinjection.annotation.ModIdentifier;
import com.dotnomi.fabricdependencyinjection.annotation.ModInject;
import com.dotnomi.fabricdependencyinjection.annotation.ModScoped;

@ModScoped
public class DataProcessor {
  private final StorageService fileStorage;

  @ModInject
  @ModIdentifier("database")
  private StorageService databaseStorage;

  @ModInject
  public DataProcessor(@ModIdentifier("file") StorageService fileStorage) {
    this.fileStorage = fileStorage;
  }

  public StorageService getFileStorage() { return fileStorage; }
  public StorageService getDatabaseStorage() { return databaseStorage; }
}

File: src/test/java/com/dotnomi/fabricdependencyinjection/kotlin/testclasses/identifier/DataProcessor.kt
package com.dotnomi.fabricdependencyinjection.kotlin.testclasses.identifier

import com.dotnomi.fabricdependencyinjection.annotation.ModIdentifier
import com.dotnomi.fabricdependencyinjection.annotation.ModInject
import com.dotnomi.fabricdependencyinjection.annotation.ModScoped

@ModScoped
class DataProcessor @ModInject constructor(@param:ModIdentifier("file") val fileStorage: StorageService) {
    @ModInject
    @ModIdentifier("database")
    lateinit var databaseStorage: StorageService
}
